src/syntax/sugar.rkt

```diff
@@ -179,65 +173,45 @@
          ;; Match guaranteed to succeed because we ran type-check first
          (define op* (apply get-parametric-operator op atypes))
          (values (cons op* args*) (impl-info op* 'otype))]
-        [(? variable?)
-         (define vrepr (context-lookup ctx expr))
-         (define repr (context-repr ctx))
-         (cond
-           [(equal? (representation-type vrepr) 'bool) (values expr vrepr)]
-           [(equal? vrepr repr) (values expr repr)]
-           [else
-            (define cast (get-cast-impl vrepr repr))
-            (unless cast
-              (raise-herbie-missing-error "conversion does not exist: ~a -> ~a"
-                                          (representation-name vrepr)
-                                          (representation-name repr)))
-            (values (list cast expr) repr)])]
+        [(? variable?) (values expr (context-lookup ctx expr))]
```

Comment:
What's the deleted code doing that we don't need to do now?

Reply:
Weird variable casting that doesn't make sense.